### PR TITLE
feat: limit tx max ancestors count

### PR DESCRIPTION
**Pull request by jjyr**
Txs with long ancestors chain affect tx pool performance. we limit max ancestors count of a single tx to resolve this issue, tx pool will reject txs which ancestors count large than the limit.

The default `max_ancestors_count` is 25.

### CODE REVIEW

File: tx-pool/src/component/proposed.rs [hunk 1]
```diff
@@ -300,22 +306,24 @@ mod tests {
...
 
         pool.add_entry(TxEntry::new(
             tx1.clone(),
             MOCK_CYCLES,
             MOCK_FEE,
             MOCK_SIZE,
             get_related_dep_out_points(&tx1, |_| None).unwrap(),
-        ));
+        ))
+        .unwrap();
```
Comment (xxuejie) [inline tx-pool/src/component/proposed.rs]: There's a lot of unwrap added here, any chance we can properly deal with the errors? Or are we sure they won't happen?

Comment (jjyr) [reply tx-pool/src/component/proposed.rs]: This is test code.

File: tx-pool/src/component/container.rs [hunk 2]
```diff
@@ -179,6 +183,18 @@ impl SortedTxMap {
...
+        if entry.ancestors_count > self.max_ancestors_count {
+            return Err(SubmitTxError::ExceededMaximumAncestorsCount);
+        }
+
+        // check duplicate tx
+        let removed_entry = if self.contains_key(&short_id) {
+            self.remove_entry(&short_id)
+        } else {
+            None
+        };
```
Comment (u2) [inline tx-pool/src/component/container.rs]: Why do you move this part of the code?

File: tx-pool/src/process/plug.rs [hunk 3]
```diff
@@ -40,12 +41,16 @@ impl Future for PlugEntryProcess {
                 match self.target {
                     PlugTarget::Pending => {
                         for entry in entries {
-                            tx_pool.add_pending(entry);
+                            if let Err(err) = tx_pool.add_pending(entry) {
+                                error!("plug entry error {}", err);
```
Comment (u2) [inline tx-pool/src/process/plug.rs]: plug?

File: tx-pool/src/component/container.rs (hunk 2 above)
Comment (jjyr) [reply tx-pool/src/component/container.rs]: If ancestors reach the limit, we should just return an error and not remove the conflict tx. Other wise we should go to the old logic.

File: tx-pool/src/process/plug.rs (hunk 3 above)
Comment (jjyr) [reply tx-pool/src/process/plug.rs]: Yes, see the file name.

**Pull request merged**